signinapp/model.py

```diff
@@ -164,13 +188,57 @@ def make(name: str, password: str, role: Role, preferred_name: str = None,
                     subteam=subteam,
                     approved=approved)
 
+    @staticmethod
+    def make_guardian(student: User, name: str, phone_number: str, email: str, preference: int = 1):
```

Comment:
It might be nice to have a `Guardian.get_from(name, phone, email)` that gets the guardian if it exists and makes them if not. That could clean up a bunch of code elsewhere.